Frameworks/CoreText/CTFrame.mm

```diff
@@ -55,9 +56,10 @@ CFRange CTFrameGetVisibleStringRange(CTFrameRef frame) {
     _CTFrame* framePtr = static_cast<_CTFrame*>(frame);
     CFIndex count = 0;
     if (framePtr) {
-        for (CFIndex i = 0; i < framePtr->_lineOrigins.size() && framePtr->_lineOrigins[i].y > 0; ++i) {
-            // Lines are ordered vertically so we can stop once we find one outside of the frame
-            count += static_cast<_CTLine*>([framePtr->_lines objectAtIndex:i])->_strRange.length;
+        for (_CTLine* line in [framePtr->_lines objectEnumerator]) {
```

Comment:
Please DO pretend it is an NSArray.

Fast enumeration is a compiler contract optimization that allows an array to return its backing store directly, in a single message dispatch.

Non-arrays can leverage it by putting up to 16 items into a provided stack buffer, potentially reducing the message sends required to `N/16`.

An object enumerator, however, due to its design must return one object at a time. You end up using all the power of the fast enumeration for object-by-object batching.

If you want to be totally type safe and not tell any lies, cast it as such: `(id<NSFastEnumeration>)x`


---
In reply to: [79686612](https://github.com/Microsoft/WinObjC/pull/989#discussion_r79686612) [](ancestors = 79686612,79682448)